nixos/modules/services/backup/borgbackup.nix

```diff
@@ -98,6 +98,34 @@ let
       inherit (cfg) startAt;
     };
 
+  mkCheckService = name: cfg:
+    let
+      userHome = config.users.users.${cfg.user}.home;
+    in nameValuePair "borgbackup-check-${name}" rec {
+      description = "Check BorgBackup repository ${name}";
+      after = [ "borgbackup-job-${name}" ];
+      conflicts = after;
+      path = with pkgs; [
+        borgbackup openssh
+      ];
+      script = ''
+        borg check
+      '';
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        # Only run when no other process is using CPU or disk
+        CPUSchedulingPolicy = "idle";
+        IOSchedulingClass = "idle";
```

Comment:
If I understand how the scheduler works, the comment is misleading; This will be ran regardless of the actual system load, but the CPU and IO priorities of the processes are going to be lower, thus meaning they're not going to have a big effect on more important processes such as CI, but they still may clog the system a bit. I think this is fine, but the comment isn't representing reality here (unless my understanding of what this option does is completely wrong, which is a real possibility)

Reply:
No, you're entirely right. I just copied both the options and the comment from the other unit, where it is also wrong.